include/dlaf/eigensolver/tridiag_solver/merge.h

```diff
@@ -0,0 +1,923 @@
+                  Matrix<SizeType, Device::CPU>(vec_size, vec_tile_size),
+                  Matrix<ColType, Device::CPU>(vec_size, vec_tile_size),
+                  Matrix<ColType, Device::CPU>(vec_size, vec_tile_size)};
+  return ws;
+}
+
+// Calculates the problem size in the tile range [i_begin, i_end]
+inline SizeType problemSize(SizeType i_begin, SizeType i_end, const matrix::Distribution& distr) {
+  SizeType nb = distr.blockSize().rows();
+  SizeType nbr = distr.tileSize(GlobalTileIndex(i_end, 0)).rows();
+  return (i_end - i_begin) * nb + nbr;
+}
+
+// Copies and normalizes a row of the `tile` into the column vector tile `col`
+//
+template <class T>
+void copyTileRowAndNormalize(SizeType row, T rho, const matrix::Tile<const T, Device::CPU>& tile,
+                             const matrix::Tile<T, Device::CPU>& col) {
+  // Negate Q1's last row if rho < 1
+  //
+  // lapack 3.10.0, dlaed2.f, line 280 and 281
+  int sign = (rho < 0) ? -1 : 1;
+  for (SizeType i = 0; i < tile.size().rows(); ++i) {
+    col(TileElementIndex(i, 0)) = sign * tile(TileElementIndex(row, i)) / std::sqrt(2);
+  }
+}
+
+DLAF_MAKE_CALLABLE_OBJECT(copyTileRowAndNormalize);
+DLAF_MAKE_SENDER_ALGORITHM_OVERLOADS(copyTileRowAndNormalize, copyTileRowAndNormalize_o)
+
+// The bottom row of Q1 and the top row of Q2. The bottom row of Q1 is negated if `rho < 0`.
+//
+// Note that the norm of `z` is sqrt(2) because it is a concatination of two normalized vectors. Hence to
+// normalize `z` we have to divide by sqrt(2). That is handled in `copyTileRowAndNormalize()`
+template <class T>
+void assembleZVec(SizeType i_begin, SizeType i_middle, SizeType i_end, pika::shared_future<T> rho_fut,
+                  Matrix<const T, Device::CPU>& mat_ev, Matrix<T, Device::CPU>& z) {
+  using pika::threads::thread_priority;
+  using pika::execution::experimental::start_detached;
+  using dlaf::internal::Policy;
+  using dlaf::internal::whenAllLift;
+
+  // Iterate over tiles of Q1 and Q2 around the split row `i_middle`.
+  for (SizeType i = i_begin; i <= i_end; ++i) {
+    // Move to the row below `i_middle` for `Q2`
+    SizeType mat_ev_row = i_middle + ((i > i_middle) ? 1 : 0);
+    GlobalTileIndex mat_ev_idx(mat_ev_row, i);
+    // Take the last row of a `Q1` tile or the first row of a `Q2` tile
+    SizeType tile_row = (i > i_middle) ? 0 : mat_ev.distribution().tileSize(mat_ev_idx).rows() - 1;
+    GlobalTileIndex z_idx(i, 0);
+    // Copy the row into the column vector `z`
+    whenAllLift(tile_row, rho_fut, mat_ev.read_sender(mat_ev_idx), z.readwrite_sender(z_idx)) |
+        copyTileRowAndNormalize(Policy<Backend::MC>(thread_priority::normal)) | start_detached();
+  }
+}
+
+// Multiply by factor 2 to account for the normalization of `z` vector and make sure rho > 0 f
+//
+template <class T>
+pika::future<T> scaleRho(pika::shared_future<T> rho_fut) {
+  return pika::dataflow(pika::unwrapping([](T rho) { return 2 * std::abs(rho); }), std::move(rho_fut));
```

Comment:
How many of the `dataflow`s in this file do you think you could change to sender adaptors? And for the ones that you can't at the moment, what is blocking you from using sender adaptors?